src/ext/language_tools.js

```diff
@@ -51,7 +53,8 @@ var snippetCompleter = {
                     caption: caption,
                     snippet: s.content,
                     meta: s.tabTrigger && !s.name ? s.tabTrigger + "\u21E5 " : "snippet",
-                    type: "snippet"
+                    type: "snippet",
+                    completerId: snippetCompleter.id
```

Comment:
while i like the idea of adding this field, i feel like it should be set in the `gatherCompletions` functions in autocomplete.js. otherwise it might be missed to be set by someone implementing it; or even setting a wrong value due to a copy/paste mistake?

Reply:
We would leave it for user to define this or not, with all default ace's completers with defined id's that shouldn't interfere with user's experience